device_doctor/lib/src/android_device.dart

```diff
@@ -93,9 +93,11 @@ class AndroidDeviceDiscovery implements DeviceDiscovery {
   @override
   Future<Map<String, List<HealthCheckResult>>> checkDevices({ProcessManager processManager}) async {
     processManager ??= LocalProcessManager();
+    final List<HealthCheckResult> defaultChecks = <HealthCheckResult>[];
```

Comment:
do we need a new variable? seems like adding check.add( .. killAdb? will be simpler.

Reply:
We want to run killAdb before running any `adb` command, including the ones in `discoverDevices`.

Reply:
Assuming all the checks are running sequentially we only need to run kill-server once at the beginning, no need to run it on every step as the adb daemon will be already running on the same user.

Reply:
That seems to contradict your first response. Anyway the comment was only cosmetic to remove a few unnecessary lines.

Reply:
Oh, I see the confusing part.
But here we need the `killAdb` before the `device for loop`, because the `discoverDevices` also calls `adb`.